x/mongo/driver/operation.go

```diff
@@ -1199,18 +1199,12 @@ func (Operation) decompressWireMessage(wm []byte) (wiremessage.OpCode, []byte, e
 	if !ok {
 		return 0, nil, errors.New("malformed OP_COMPRESSED: missing compressor ID")
 	}
-	compressedSize := len(wm) - 9 // original opcode (4) + uncompressed size (4) + compressor ID (1)
```

Comment:
We don't have to calculate the length as long as the above `ReadCompressedOriginalOpCode`, `ReadCompressedUncompressedSize`, and `ReadCompressedCompressorID` are correct. `compressedSize` must be strictly equal to `len(rem)` because it is where `rem` comes from. Therefore, the `ReadCompressedCompressedMessage` is unnecessary as well.